environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java

```diff
@@ -152,6 +154,11 @@ static WeldContainer initialize(String id, WeldManager manager, Bootstrap bootst
         return weldContainer;
     }
 
+    private static boolean isShutdownHookNeeded() {
+        String property = AccessController.doPrivileged(new GetSystemPropertyAction(Weld.SHUTDOWN_HOOK_SYSTEM_PROPERTY));
+        return property == null || Boolean.FALSE.equals(Boolean.valueOf(property));
```

Comment:
@astefanutti When I think about it... it will be better to use `org.jboss.weld.se.shutdownHook` (and follow the Weld config convention) and change this logic. So that one would have to use `org.jboss.weld.se.shutdownHook=false` to skip the registration.